### PR TITLE
Fix: Lineage 

### PR DESCRIPTION
**Pull request by vchan**
This PR
- Normalizes table names so the UI can correctly match them with their model's fqn.
- Handles the cases where a lineage node's expression are unions and CTEs.

### CODE REVIEW

File: web/server/api/endpoints/lineage.py [hunk 1]
```diff
@@ -27,16 +28,25 @@ def _get_table(node: Node, dialect: t.Optional[DialectType] = None) -> t.Optiona
     table: t.Union[exp.Table, str] = node.alias or node.name
     if isinstance(node.expression, exp.Table):
         table = node.expression
+    elif isinstance(node.expression, exp.Alias):
+        table = node.expression.this.table
 
     try:
-        return exp.table_name(table, identify=True, dialect=dialect)
+        return normalize_model_name(table, None, dialect=dialect)
```
Comment (tobymao) [inline web/server/api/endpoints/lineage.py:35]: why are you passing None?

File: web/server/api/endpoints/lineage.py [hunk 2]
```diff
@@ -27,16 +28,25 @@ def _get_table(node: Node, dialect: t.Optional[DialectType] = None) -> t.Optiona
...
     except sqlglot.errors.ParseError:
         # Cannot extract table from node. One reason this can happen is node is
         # '*' because a model selects * from an external model for which we do
         # not know the schema.
         return None
 
 
+def _get_column_name(node: Node) -> str:
+    if isinstance(node.expression, exp.Alias):
```
Comment (tobymao) [inline web/server/api/endpoints/lineage.py:44]: why not node.expression.alias_or_name?

File: web/server/api/endpoints/lineage.py [hunk 3]
```diff
@@ -106,25 +134,25 @@ async def column_lineage(
...
-            column_name = exp.to_column(node.name).name
-        if column_name in graph.get(node_name, []):
+        column_name = _get_column(node, dialect)
+        if column_name in graph.get(table, []):
             continue
 
         # At this point node_name should be fqn/normalized/quoted
-        graph[node_name] = {
+        graph[table] = {
             column_name: LineageColumn(
```
Comment (mykalmax) [inline web/server/api/endpoints/lineage.py:152]: i think , we need to append here as well

File: web/server/api/endpoints/lineage.py (hunk 1 above)
Comment (vchan) [reply web/server/api/endpoints/lineage.py]: Sorry, the PR wasn't done yet. I'm passing in default catalog now.

File: tests/web/test_lineage.py [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+
+    response = client.get("/api/lineage/foo/col")
+    assert response.status_code == 200, response.json()
+    response_json = response.json()
+    assert response_json['"foo"']["col"]["models"] == {'"my_cte1"': ["col"]}
+    assert response_json['"my_cte1"']["col"]["models"] == {
+        '"my_cte2"': ["col"],
+        '"external_table1"': ["col"],
+    }
+    assert response_json['"my_cte2"']["col"]["models"] == {'"bar"': ["col"]}
```
Comment (tobymao) [inline tests/web/test_lineage.py:434]: wouldn't it be easier just to check the whole json response as opposed to manually checking each snippet

Comment (vchan) [reply tests/web/test_lineage.py:434]: The whole json response can get quite cluttered with the expression and source sql.

**Pull request merged**